core/rio/hdt/src/main/java/org/eclipse/rdf4j/rio/hdt/part/HDTBitmap.java

```diff
@@ -0,0 +1,79 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Eclipse RDF4J contributors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Distribution License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *******************************************************************************/
+package org.eclipse.rdf4j.rio.hdt.part;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.zip.CheckedInputStream;
+
+import org.eclipse.rdf4j.rio.hdt.util.CRC32;
+import org.eclipse.rdf4j.rio.hdt.util.CRC8;
+import org.eclipse.rdf4j.rio.hdt.util.VByte;
+
+/**
+ *
+ * @author Bart Hanssens
+ */
+public class HDTBitmap extends HDTPart {
+	public final static int BITMAP1 = 1;
+
+	private int bits;
+	private byte[] buffer;
+
+	/**
+	 * Get bit
+	 * 
+	 * @param i
+	 * @return 0 or 1
+	 */
+	public int get(int i) {
+		assert i < bits : "Bitmap out of range";
```

Comment:
How important is this assertion? Not familiar with recent RDF4J coding standards, but using an always-on check and exception rather than a variable assertion is more generally accepted.

Reply:
Good catch. I think we've stepped away from using assertions. They still lurk in the code base in places, but I'd prefer to gradually get rid of them in favor of exceptions.

Reply:
Some background for assertions vs. exceptions. Assertions will only run if you enable assertions (like in a test), unlike exceptions. Assertions in small methods also hinder inlining (method becomes too big). 

That said, assertions are a helpful tool when debugging code. I find them especially useful while developing new data structures.

Another use case for assertions is when the check is semi important, but takes a lot of resources. I've once used it to discover resource leaks on a db connection where I had to use java reflection, which is very slow. The resource leak was important to catch, but I was fairly sure that one of my integration tests would catch it and didn't really want to crash in production either.

Reply:
I'll just remove them, the main use was for catching some off-by-one errors in development